### PR TITLE
Fix/Sent message progress infinite

### LINKED ISSUE #247: Sent message progress is infinite.
Sometimes is hard to reproduce.

### PR DESCRIPTION
**Pull request by adrian09h**
# Submit a pull request (close #247 )
- Remove `MessageStatus`
- Message.setSyncStatus(Sync.SYNCED) on `ChatChannelEventHandler` and `ChatEventHandler`

```java
final void dispatchEvent(Client client, Event event) {   
        switch (event.getType()) {
            ...            
            case MESSAGE_NEW:
                event.getMessage().setSyncStatus(Sync.SYNCED);
                dispatchChannelEvent(client, event, this::onMessageNew);
                break;
            case MESSAGE_UPDATED:
                event.getMessage().setSyncStatus(Sync.SYNCED);
                dispatchChannelEvent(client, event, this::onMessageUpdated);
                break;
            case MESSAGE_DELETED:
                event.getMessage().setText(Constant.MESSAGE_DELETED);
                dispatchChannelEvent(client, event, this::onMessageDeleted);
                break;
           ...
        }
}
```

### CODE REVIEW

File: library/src/main/java/com/getstream/sdk/chat/rest/core/ChatChannelEventHandler.java [hunk 1]
```diff
@@ -61,12 +63,15 @@ public final void dispatchEvent(Event event) {
...
             case MESSAGE_NEW:
+                event.getMessage().setSyncStatus(Sync.SYNCED);
                 onMessageNew(event);
                 break;
             case MESSAGE_UPDATED:
+                event.getMessage().setSyncStatus(Sync.SYNCED);
                 onMessageUpdated(event);
                 break;
             case MESSAGE_DELETED:
+                event.getMessage().setText(Constant.MESSAGE_DELETED);
```
Comment (elevenetc) [inline library/src/main/java/com/getstream/sdk/chat/rest/core/ChatChannelEventHandler.java]: `Constant.MESSAGE_DELETED` should be moved to `strings.xml`

File: library/src/main/java/com/getstream/sdk/chat/rest/core/ChatChannelEventHandler.java [hunk 2]
```diff
@@ -61,12 +63,15 @@ public final void dispatchEvent(Event event) {
...
                 break;
             case MESSAGE_NEW:
+                event.getMessage().setSyncStatus(Sync.SYNCED);
                 onMessageNew(event);
                 break;
             case MESSAGE_UPDATED:
+                event.getMessage().setSyncStatus(Sync.SYNCED);
                 onMessageUpdated(event);
                 break;
             case MESSAGE_DELETED:
```
Comment (elevenetc) [inline library/src/main/java/com/getstream/sdk/chat/rest/core/ChatChannelEventHandler.java]: Should be status also updated?
```
event.getMessage().setSyncStatus(Sync.SYNCED);
```

Comment (elevenetc) [reply library/src/main/java/com/getstream/sdk/chat/rest/core/ChatChannelEventHandler.java]: The same about `ChatEventHandler`

File: library/src/main/java/com/getstream/sdk/chat/rest/core/ChatChannelEventHandler.java (hunk 1 above)
Comment (elevenetc) [reply library/src/main/java/com/getstream/sdk/chat/rest/core/ChatChannelEventHandler.java]: The same about `ChatEventHandler`

File: library/src/main/java/com/getstream/sdk/chat/rest/core/Client.java [hunk 3]
```diff
@@ -1109,11 +1109,15 @@ public void updateMessage(@NonNull Message message,
 
             @Override
             public void onResponse(Call<MessageResponse> call, Response<MessageResponse> response) {
+                message.setSyncStatus(SYNCED);
+                if (response.body() != null && response.body().getMessage() != null)
```
Comment (elevenetc) [inline library/src/main/java/com/getstream/sdk/chat/rest/core/Client.java]: We need to log if there is no `body` and `message`. That's an exceptional case. And call `callback.onError`.
Also could be wrapped into some `ResponseUtilities` like as it is repeated:
```java
ResponseUtilities.setMessageStatus(response, SYNCED);
```

File: library/src/main/java/com/getstream/sdk/chat/StreamChat.java [hunk 4]
```diff
@@ -39,6 +33,7 @@
     private static Integer lastUnreadChannels;
     private static boolean lifecycleStopped;
     private static boolean userWasInitialized;
+    private static Context mContext;
```
Comment (elevenetc) [inline library/src/main/java/com/getstream/sdk/chat/StreamChat.java]: Let's drop `m` prefix. We're not using it elsewhere, so no reason to have it here.

File: library/src/main/java/com/getstream/sdk/chat/adapter/MessageListItemViewHolder.java [hunk 5]
```diff
@@ -290,33 +290,24 @@ private void configDeliveredIndicator() {
...
                 pb_deliver.setVisibility(View.VISIBLE);
                 iv_deliver.setVisibility(View.GONE);
                 break;
-            case RECEIVED:
+            case Sync.SYNCED:
                 pb_deliver.setVisibility(View.GONE);
                 iv_deliver.setVisibility(View.VISIBLE);
                 break;
-            case FAILED:
+            case Sync.IN_MEMORY:
```
Comment (elevenetc) [inline library/src/main/java/com/getstream/sdk/chat/adapter/MessageListItemViewHolder.java]: Empty cases should also have breaks. And would be great to have a comment if there is a reason why it's empty.

Comment (adrian09h) [reply library/src/main/java/com/getstream/sdk/chat/adapter/MessageListItemViewHolder.java]: > Empty cases should also have breaks. And would be great to have a comment if there is a reason why it's empty.

`Sync.IN_MEMORY` should also hide `pb_deliver` and pb_deliver like `Sync.LOCAL_FAILED` case.
So it has not breaks.

**Pull request merged**